### DISCUSSION TITLE
Consent reset,  if data is modified by the user?

### DISCUSSION THREAD

**Discussion by rahul101001000** [Sunbird-RC/community#66] [Sunbird RC Technical Specification Feedback]
Is there a concept of consent reset if a user changes updates an attribute?

e.g. user changes email address, is there an option to specify whether consent for email will need to be given again? (Options could be one time only, every change). Thanks! 

@pramodkvarma

**Comment by pramodkvarma** [comment]
If the user has option to change visibility (based on the attribute config called "user-controlled" as @coolbung and @dileepbapat suggested), then she/he should be able to change it any time. If user updates the attribute, UI can show that checkbox/toggle button again as a best practice.

**Reply by rahul101001000** [reply]
That discussion deals with changing of attribute visibility between internal, private with consent, and public. 

This one refers to an attribute that continues to be private but the value of which is changed by the user. Let's say I update my email from a@z.com to b@z.com. I would like everyone who I've granted consent to access this attribute in the past, to ask for access again. 

Essentially this would be a new attribute config for consent reset. 

IIRC Trust Id framework has this part of their spec. See table 3 in this [attribute profile doc](https://www.digitalidentity.gov.au/sites/default/files/2021-06/tdif-06d-attribute-profile-release-4-V1.1_0.pdf)

**Reply by pramodkvarma** [reply]
Thanks for the pointer. Will check out. In general, it is not a good idea to push to everyone in the past who had accessed the profile. Next time user interacts with service provider, they can do a refresh.

**Comment by rahul101001000** [comment]
Yes makes sense. If the user wants to explicitly change consent given, they
can always revoke access to a certain information user.
